opentracing-api/src/main/java/io/opentracing/Format.java

```diff
@@ -11,9 +11,16 @@
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  */
-package io.opentracing.propagation;
+package io.opentracing;
 
-/** work in progress */
-public interface BinaryReader {
-    // TODO
+/**
+ * Format identifies a reader (R) and Writer (W) pair used by Tracer.inject() and Tracer.extract() to encode/decode SpanContext instances.
+ *
+ * Nearly all OpenTracing use cases are satisfied by the BuiltinFormats, but Format provides extensibility when custom carriers are warranted.
+ *
+ * @see BuiltinFormats
+ * @see io.opentracing.Tracer#inject(SpanContext, Format, Object)
+ * @see io.opentracing.Tracer#extract(Format, Object)
+ */
+public interface Format<R, W> {
```

Comment:
I guess the builtin formats could also just be public static members inside here...

```
public interface Format<R, W> {
    public final static Format<String, String> TextMap = new Builtins<>();

    class Builtins<R, W> implements Format<R, W> {
    }
}
```

Thoughts?